src/jennifer/config.cr

```diff
@@ -4,139 +4,179 @@ require "logger"
 module Jennifer
   class Config
     CONNECTION_URI_PARAMS = [:max_pool_size, :initial_pool_size, :max_idle_pool_size, :retry_attempts, :checkout_timeout, :retry_delay]
-    STRING_FIELDS = {:user, :password, :db, :host, :adapter, :migration_files_path, :schema, :structure_folder}
-    INT_FIELDS    = {:port, :max_pool_size, :initial_pool_size, :max_idle_pool_size, :retry_attempts}
-    FLOAT_FIELDS  = [:checkout_timeout, :retry_delay]
 
-    macro define_fields(const, default)
-      {% for field in @type.constant(const.stringify) %}
-      @@{{field.id}} = {{default}}
+    class ConfigInstance
+      STRING_FIELDS = {:adapter, :user, :password, :db, :host, :adapter, :migration_files_path, :schema, :structure_folder, :key}
+      INT_FIELDS    = {:port, :max_pool_size, :initial_pool_size, :max_idle_pool_size, :retry_attempts}
+      FLOAT_FIELDS  = [:checkout_timeout, :retry_delay]
 
-      def self.{{field.id}}=(value)
-        @@{{field.id}} = value
+      macro define_fields(const, default)
```

Comment:
as we moving to storing config properties in objects we will have no need in such fields definition - can be removed

Reply:
but anyway such attribute as `@logger` should be explicitly defined anyway